runtime/src/main/java/org/corfudb/protocols/wireprotocol/orchestrator/OrchestratorRequest.java

```diff
@@ -0,0 +1,49 @@
+package org.corfudb.protocols.wireprotocol.orchestrator;
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+
+import org.corfudb.format.Types.OrchestratorRequestType;
+import org.corfudb.protocols.wireprotocol.ICorfuPayload;
+
+import static org.corfudb.format.Types.OrchestratorRequestType.ADD_NODE;
+
+/**
+ * A message container that encapsulates all the orchestrator's
+ * requests.
+ *
+ * Created by Maithem on 10/25/17.
+ */
+
+public class OrchestratorRequest implements ICorfuPayload<OrchestratorRequest> {
+
+    @Getter
+    public Request request;
+
+    public OrchestratorRequest(Request request) {
+        this.request = request;
+    }
+
+    public OrchestratorRequest(ByteBuf buf) {
+        OrchestratorRequestType requestType = OrchestratorRequestType.forNumber(buf.readInt());
+        byte[] bytes = new byte[buf.readInt()];
+        buf.readBytes(bytes);
+        request = mapRequest(requestType, bytes);
+    }
+
+    @Override
+    public void doSerialize(ByteBuf buf) {
+        buf.writeInt(request.getType().getNumber());
+        byte[] bytes = request.getSerialized();
+        buf.writeInt(bytes.length);
+        buf.writeBytes(bytes);
+    }
+
+    static Request mapRequest(OrchestratorRequestType type, byte[] payload) {
+        if (type.equals(ADD_NODE)) {
+            return new AddNodeRequest(payload);
+        } else {
+            throw new RuntimeException("Unknown Orchestrator Type");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)